core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -174,11 +173,26 @@ private class SystemFunctionalityListener implements DatabaseEventListener {
 
         @Override
         public void databaseDrop(DatabaseEventContext eventContext) {
-
-            forEachSystemLabel((tx, label) -> {
-                tx.findNodes(label, database.name(), eventContext.getDatabaseName())
-                        .forEachRemaining(Node::delete);
-            });
+            final var dbName = eventContext.getDatabaseName();
+            if (!Objects.equals(dbName, db.databaseName())) {
+                if (!db.isAvailable()) {
+                    userLog.warn(
+                            "Database %s not available, skipping apoc trigger cleanup of database %s.",
+                            db.databaseName(), dbName);
+                    return;
+                }
+                try (final var tx = db.beginTx()) {
+                    for (Label label : SystemLabels.values()) {
+                        tx.findNodes(label, database.name(), eventContext.getDatabaseName())
+                                .forEachRemaining(Node::delete);
+                    }
+                    tx.commit();
+                } catch (Exception e) {
+                    userLog.error(
+                            "Failed to cleanup apoc triggers during database drop of %s: %s".formatted(dbName, e), e);
+                    throw e;
```

Comment:
Should we even swallow this exception?

Reply:
Yeah, I think swallow it

Reply:
Makes sense, updated the PR to ignore this error.